analyzers/tests/SonarAnalyzer.UnitTest/TestCases/TimestampsShouldNotBeUsedAsPrimaryKeys.cs

```diff
@@ -0,0 +1,139 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+using DateTimeAlias = System.DateTime;
+using KeyAttributeAlias = System.ComponentModel.DataAnnotations.KeyAttribute;
+
+class TemporalTypes
+{
+    class Entity
+    {
+        public DateTime EntityId { get; set; }          // Noncompliant {{Timestamps should not be used as primary keys}}
+        //     ^^^^^^^^
+    }
+
+    class DateTimeKey
+    {
+        public DateTime Id { get; set; }                // Noncompliant
+    }
+
+    class DateTimeOffsetKey
+    {
+        public DateTimeOffset Id { get; set; }          // Noncompliant
+    }
+
+    class TimeSpanKey
+    {
+        public TimeSpan Id { get; set; }                // Noncompliant
+    }
+
+    class DateTimeNoKey
+    {
+        public DateTime Identifier { get; set; }        // Compliant - only Id and [class-name]Id is recognized as key by Entity Framework
+    }
+
+    class DifferentCasing
+    {
+        public DateTime ID { get; set; }                // Noncompliant
```

Comment:
Educational, is the check for EntityFramework case insensitive? Will it really pick up ID or is it picking up only Id?

Reply:
Yes, I checked it in a small test project. It's case insensitive for both EF6 and EF Core.